sys/net/gnrc/network_layer/ipv6/nib/_nib-internal.h

```diff
@@ -0,0 +1,476 @@
+ */
+typedef struct {
+    _nib_t *next_hop;           /**< next hop to destination */
+    ipv6_addr_t pfx;            /**< prefix to the destination */
+    unsigned pfx_len;           /**< prefix-length of _nib_t::dst */
+} _nib_dst_t;
+
+/**
+ * @brief   Interface specific information for Neighbor Discovery
+ */
+typedef struct {
+#if GNRC_IPV6_NIB_CONF_ARSM
+    uint32_t reach_time_base;           /**< base for random reachable time calculation */
+    uint32_t reach_time;                /**< reachable time (in ms) */
+    uint32_t retrans_time;              /**< retransmission time (in ms) */
+#endif
+#if GNRC_IPV6_NIB_CONF_ROUTER || defined(DOXYGEN)
+    /**
+     * @brief   timestamp in milliseconds of last unsolicited router advertisement
+     *
+     * @note    Only available if @ref GNRC_IPV6_NIB_CONF_ROUTER
+     */
+    uint32_t last_ra;
+#endif
+    kernel_pid_t pid;                   /**< identifier of the interface */
+#if GNRC_IPV6_NIB_CONF_ROUTER || defined(DOXYGEN)
+    /**
+     * @brief   number of unsolicited router advertisements sent
+     *
+     * This only counts up to the first @ref NDP_MAX_INIT_RA_NUMOF on interface
+     * initialization. The last @ref NDP_MAX_FIN_RA_NUMOF of an advertising
+     * interface are counted from UINT8_MAX - @ref NDP_MAX_FIN_RA_NUMOF + 1.
+     *
+     * @note    Only available if @ref GNRC_IPV6_NIB_CONF_ROUTER
+     */
+    uint8_t ra_sent;
+#endif
+    uint8_t rs_sent;                    /**< number of unsolicited router solicitations scheduled */
+    /**
+     * @brief   number of unsolicited neighbor advertisements scheduled
+     */
+    uint8_t na_sent;
+} _nib_iface_t;
+
+/**
+ * @brief   Mutex for locking the NIB.
+ */
+extern mutex_t _nib_mutex;
+
+/**
+ * @brief   Initializes NIB internally
+ */
+void _nib_init(void);
+
+/**
+ * @brief   Gets interface identifier from a NIB entry
+ *
+ * @param[in] nib   A NIB entry
+ *
+ * @return  The NIB entry's interface identifier
+ */
+static inline unsigned _nib_get_if(const _nib_t *nib)
+{
+    return (nib->info & _NIB_IF_MASK) >> _NIB_IF_POS;
+}
+
+/**
+ * @brief   Sets interface for a NIB entry
+ *
+ * @param[in,out] nib   A NIB entry
+ * @param[in] iface     An interface identifier
+ */
+static inline void _nib_set_if(_nib_t *nib, unsigned iface)
+{
+    assert(iface <= 0x1f);
+    nib->info &= ~(_NIB_IF_MASK);
+    nib->info |= ((iface << _NIB_IF_POS) & _NIB_IF_MASK);
+}
+
+/**
+ * @brief   Creates or gets an existing on-link entry by address
+ *
+ * @pre     `(addr != NULL)`
+ *
+ * @param[in] addr  An IPv6 address. May not be NULL.
+ *                  *May also be a global address!*
+ * @param[in] iface The interface to the node.
+ *
+ * @return  A new or existing on-link entry with _nib_t::ipv6 set to @p addr.
+ * @return  NULL, if no space is left.
+ */
+_nib_t *_nib_alloc(const ipv6_addr_t *addr, unsigned iface);
+
+/**
+ * @brief   Clears out a NIB entry (on-link version)
+ *
+ * @param[in] nib  An entry
```

Comment:
It isn't really an in-out-parameter... it's more the object or context of the call. But I guess you can call that an in-out-parameter from a certain point of view.